bonelab/cli/PointPicker_implement.py

```diff
@@ -0,0 +1,327 @@
+    smoothImg.SetNumberOfIterations(5)
+    smoothImg.SetRelaxationFactor(0.2)
+    smoothImg.FeatureEdgeSmoothingOff()
+    smoothImg.BoundarySmoothingOn()
+    smoothImg.Update()
+    
+    # Calculate normals for triangle strips
+    normals = vtk.vtkPolyDataNormals()
+    normals.SetInputConnection(smoothImg.GetOutputPort())
+    normals.ComputePointNormalsOn()
+    normals.ComputeCellNormalsOn()
+    normals.ConsistencyOn()
+    normals.Update()
+    
+    # create a dict of preset colors (i.e. bone, skin, blood, etc)
+    colors = {}
+    colors = {'skin':(0.90, 0.76, 0.6), 
+              'bone':(0.83, 0.8, 0.81), 
+              'red':(0.65, 0.1, 0.1), 
+              'green':(0, 0.9, 0.2), 
+              'purple':(0.7, 0.61, 0.85)}
+
+    # initialise mapper of opaque 3D image
+    mapper3DOpaque = vtk.vtkPolyDataMapper()
+    mapper3DOpaque.SetInputConnection(normals.GetOutputPort())
+    mapper3DOpaque.ScalarVisibilityOff()
+
+    # initialise the opaque actor
+    actor3DOpaque = vtk.vtkActor()
+    actor3DOpaque.SetMapper(mapper3DOpaque)
+    actor3DOpaque.GetProperty().SetOpacity(1.0)
+    actor3DOpaque.GetProperty().SetColor(colors['bone'])
+
+    # create the renderer and add actors
+    renderer = vtk.vtkRenderer()
+    renderer.AddActor(actor3DOpaque)
+    
+    # create the render window
+    renderWindow = vtk.vtkRenderWindow()
+    renderWindow.AddRenderer(renderer)
+    renderWindow.SetSize(1000, 1000)
+    
+    # Connect an interactor to the image viewer
+    interactor = vtk.vtkRenderWindowInteractor()
+    interactor.SetRenderWindow(renderWindow)
+    interactor.SetInteractorStyle(vtk.vtkInteractorStyleTrackballCamera())
+
+    # =============================================================================
+    # Press 'p' to add a point to the display and to the existing points dictionaries
+    # Initialised on key press event in the interactor
+    # =============================================================================    
+    def pickCell(interactor, event):
+        
+        keyPress = interactor.GetKeySym()
+        
+        if keyPress == 'p' or keyPress =='P':
+            
+            # get cursor position when 'p' is pressed
+            x, y = interactor.GetEventPosition()
+            
+            # determine where the cursor is when 'p' is pressed
+            cellPicker = vtk.vtkCellPicker()
+            cellPicker.PickFromListOn()
+            cellPicker.AddPickList(actor3DOpaque)
+            cellPicker.SetTolerance(0.0001)
+            cellPicker.Pick(x, y, 0, renderer)
+            
+            points = cellPicker.GetPickedPositions()
+            numPoints = points.GetNumberOfPoints()
+            if numPoints < 1:
+                return()
+            i, j, k = points.GetPoint(0)
+            
+            # create a sphere at the location of the mouse cursor
+            sphere = vtk.vtkSphereSource()
+            sphere.SetRadius(0.005 * (reader.GetOutput().GetDimensions()[1]))
```

Comment:
Nice trick here. I know I recommended this before, but maybe it makes more sense to set it at the command line. You can allow the user to specify a number (say, radius in mm) and if it is negative, calculate it yourself using this code.